app.py

```diff
@@ -0,0 +1,437 @@
+                refund_fields[environment]["customer_name"]
+            ],
+            refund_fields[environment]["event_name"]: record_data[
+                refund_fields[environment]["event_name"]
+            ],
+            refund_fields[environment]["type"]: record_data[
+                refund_fields[environment]["type"]
+            ],
+            refund_fields[environment]["banner_reservations_lpb"]: lpb_connection_id,
+            refund_fields[environment]["banner_reservations_ots"]: ots_connection_id,
+            refund_fields[environment]["sub_description"]:  record_data[
+                refund_fields[environment]["sub_description"]
+            ],
+        }
+    )
+
+
+def create_message_json(
+    environment, citybase_id, today_date, knack_invoice, payment_status
+):
+    """
+    :param environment: "uat" or "prod" depending on which endpoint calls this function
+    :param citybase_id: citybase transaction id
+    :param today_date: mm/dd/YYYY H:M datetime string
+    :param knack_invoice: info from citybase payload
+    :param payment_status: info from citybase payload
+    :return: json object to insert in knack citybase_messages table
+    """
+    return json.dumps(
+        {
+            field_maps[environment]["messages_invoice_id"]: knack_invoice,
+            field_maps[environment]["messages_connected_invoice"]: knack_invoice,
+            field_maps[environment]["messages_created_date"]: today_date,
+            field_maps[environment]["messages_status"]: payment_status,
+            field_maps[environment]["messages_citybase_id"]: citybase_id,
+        }
+    )
+
+
+def update_parent_reservation(environment, knack_record_id, today_date):
+    """
+    Uses a knack_record_id to look up the transaction in knack, then finds the appropriate connection field
+    And sends payload to knack, marking parent reservation payment received status as TRUE
+    """
+    record_response = requests.get(
+        f"{KNACK_API_URL}{TRANSACTIONS_OBJECT_ID}/records/{knack_record_id}",
+        headers=headers[environment],
+    )
+    record_data = record_response.json()
+    if record_data[field_maps[environment]["ots_connection_field"]]:
+        # get parent record id
+        parent_record_response = record_data[
+            field_maps[environment]["ots_connection_field"] + "_raw"
+        ]
+        parent_record_id = parent_record_response[0]["id"]
+        ots_payload = json.dumps(
+            {
+                field_maps[environment]["ots_application_status"]: "Approved",
+                field_maps[environment]["ots_paid_status"]: True,
+                field_maps[environment]["ots_payment_date"]: today_date,
+            }
+        )
+        parent_update_response = requests.put(
+            f"{KNACK_API_URL}{OTS_OBJECT_ID}/records/{parent_record_id}",
+            headers=headers[environment],
+            data=ots_payload,
+        )
+        print(parent_update_response)
```

Comment:
I've had hit-or-miss results in the past trying to get messages to appear from within a flask app using the vanilla `print()` function. You may want to consider the logging methods built into the flask object itself, like `app.logger.info("Parent update response: " + str(parent_update_response))` or something similar. Doing this can also let you separate informational output from errors if the deployment environment is set to monitor for errors being emitted from the application.

Reply:
This is really helpful, thank you -- the print statements were helpful when I was running it locally, I will switch over to the logging method